lib/email_parser/parser.rb

```diff
@@ -324,13 +324,31 @@ def self.strip_html(body, headers)
       return nil if body.nil?
 
       doc = Nokogiri::HTML(body)
-      # TODO: Handle non-gmail html sources
-      if headers.keys.include?('X-Google-Smtp-Source')
-        quoted_content = doc.xpath("//div[@class='gmail_quote']").first
-        quoted_content&.remove
 
-        signature_content = doc.xpath("//div[@class='gmail_signature']").first
-        signature_content&.remove
+      # remove quoted content
+      remove_element_onwards(doc.xpath("//div[@class='gmail_quote']").first) # gmail clients
+      remove_element_onwards(doc.xpath("//div[@id='appendonsend']").first) # office365 clients
+      remove_element_onwards(doc.xpath("//blockquote").first) ## apple clients
+
+      # remove "On Apr 12, 2019, at 12:42 PM, Amit Koren <amit@intrologic.com> wrote:"
+      doc.xpath('//div').find { |node| /on.*wrote:$/i.match(node.text) }&.remove
+
+      # remove signature
+      remove_element_onwards(doc.xpath("//div[@class='gmail_signature']").first) # gmail clients
+      remove_element_onwards(doc.xpath("//div[@id='AppleMailSignature']").first) # ios mail app
+      remove_element_onwards(doc.xpath("//div[@id='Signature']").first) # office365 clients
+
```

Comment:
Should we also remove the forwarded email if any? Or the `Sent with my iphone part` (not sure if that's part of the Signature div)

Reply:
I think the forwarded email will be removed above in the quoted content section.  "Sent with my iphone" was part of the AppleMailSignature div

Reply:
`<div id="AppleMailSignature" dir=3D"ltr">Sent from my iPhone</div>`